src/codemagic/cli/argument/typed_cli_argument.py

```diff
@@ -55,7 +55,10 @@ def from_environment_variable_default(cls) -> Optional['TypedCliArgument[T]']:
             return None
         elif cls.environment_variable_key not in os.environ:
             return None
-        return cls(os.environ[cls.environment_variable_key], from_environment=True)
+        try:
+            return cls(os.environ[cls.environment_variable_key], from_environment=True)
+        except argparse.ArgumentTypeError as ate:
+            raise ValueError(str(ate)) from ate
```

Comment:
This is the gist of the bugfix. `from_environment_variable_default` is called from `Argument.from_args` which handles `ValueErrors` and has context to properly raise `argparse.ArgumentError` that can later be properly handled to produce nice error message.